checker/raw_result.go

```diff
@@ -186,7 +188,7 @@ type ContributorsData struct {
 // VulnerabilitiesData contains the raw results
 // for the Vulnerabilities check.
 type VulnerabilitiesData struct {
-	Vulnerabilities []clients.Vulnerability
+	OSVVulnerabilities osvmodels.VulnerabilityResults
```

Comment:
Generally, we want this structure to be agnostic to the database we query, so that if we ever add other databases (CVEs), we'd be able to do that. This means you'd need to translate some fields to the `clients.Vulnerability` structure.

An alternative - what you propose - is to expose OSV format directly. I think this also has benefits: users familiar with OSVs already know how to parse the results. But I'm not sure if we want to expose that level of granularity within Scorecard, since users who care about this can just query OSV directly or use a URL Scorecard can provide. 

One downside of exposing OSV directly is that if you bump the major version, Scorecard will need to bump its version too.

So I think I'm in favor of keeping the results database agnostic, ie keep `clients.Vulnerability`. The ID should already be unique `OSV-XXX` vs `CVE-XXX`. 

@azeemshaikh38 wdut?

Reply:
That makes sense to me, I have reverted the change here so that the interface is the same to avoid the major version bump. We can always come back to this later if we want to expose more details.

Reply:
+1. The way I visualized this is, we'd extend the `VulnerablitiesData` struct such that it includes a mapping of project name and its associated vulnerability. A list of such a mapping can easily encapsulate both the project's direct and indirect vulnerabilities. Then an underlying implementation (like OSV Scanner) would fill in this information to the best of it ability.